src/labelle/lib/devices/device_config.py

```diff
@@ -50,19 +55,57 @@ def matches_device_id(self, idValue: int) -> bool:
         else:
             return False
 
-    def get_tape_print_margins(self, tapeSizeMM: int) -> tuple[int, int]:
+    def get_tape_print_size_and_margins_px(
+        self, tapeSizeMM: int
+    ) -> tuple[int, int, int]:
         """Get print margins for a specific tape size.
 
         :param tapeSizeMM: Tape size in mm
-        :return: Margins tuple, None if not supported
+        :return: Margins tuple in pixels [ActivePixels,TopMarginPx,BottomMarginPx]
         """
         if tapeSizeMM in self.supportedTapeSizes:
-            if tapeSizeMM in self.marginsPerTape:
-                # Return specified margins
-                return self.marginsPerTape[tapeSizeMM]
+            # Calculate the pixels per mm for this printer
+            # Example: printhead of 128 Pixels, distributed over 18 mm of active area.
+            #   Makes 7.11 pixels/mm
+            printPPmm: float = (
+                self.printHeadSizePixels / self.printHeadActiveAreaMillimeters
+            )
+
+            # Calculate usable tape width (*2 for top and bottom)
+            usableTapeWidth: float = tapeSizeMM - (
```

Comment:
Please make this a property, not a calculated variable inside a function.

Reply:
The issue is that you need the tape with as input to calculate this. 
am I missing something?

Reply:
Moved to the labeler class and made it a property